smtp/smtpc.c

```diff
@@ -440,10 +452,21 @@ smtpc_execute(struct smtpc_request *req, double timeout)
 		++env->stat.failed_requests;
 		smtpc_request_delete(req);
 		return -1;
+	case CURLE_SEND_ERROR:
+	case CURLE_RECV_ERROR:
+		curl_easy_getinfo(req->easy, CURLINFO_RESPONSE_CODE, &longval);
+		req->status = (int) longval;
+		if (strcmp(req->smtp_error, "response reading failed") == 0) {
+			req->status = -1;
+		}
+		strcat(req->smtp_error, " (SMTP error)");
+		req->reason = req->smtp_error;
+		++env->stat.failed_requests;
+		break;
 	default: {
 		char error_msg[256];
 		curl_easy_getinfo(req->easy, CURLINFO_OS_ERRNO, &longval);
-		snprintf(error_msg, sizeof(error_msg), "SMTP error %i (os errno %li)", req->code, longval);
+		snprintf(error_msg, sizeof(error_msg), "CURL error %i (os errno %li)", req->code, longval);
```

Comment:
The codes which are represented in this line have nothing to do with SMTP codes. So that must be either some CURL error code with its value or I would just print OS error code here.